datastore.go

```diff
@@ -142,9 +152,10 @@ func (d *Datastore) Delete(key ds.Key) error {
 }
 
 func (d *Datastore) Query(q dsq.Query) (dsq.Results, error) {
-	txn := d.NewTransaction(true)
-	defer txn.Discard()
```

Comment:
This `defer` is problematic with badger master, as it discards the txn while the iterator is active.

Reply:
This is the error this PR solves:

```
panic: Unclosed iterator at time of Txn.Discard. [recovered]
	panic: Unclosed iterator at time of Txn.Discard.
```